tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/RecordEntry.cs

```diff
@@ -26,16 +26,51 @@ public class RecordEntry
 
         public RequestOrResponse Response { get; set; } = new RequestOrResponse();
 
-        public string RequestUri { get; set; }
+        private string _requestUri;
+        public string RequestUri
+        {
+            get { return this._requestUri; }
+            set {
+                // If the requestUri is being modified, set the flag to true
+                if (DebugLogger.CheckLogLevel(LogLevel.Debug) && this._requestUri != value) requestUriIsModified = true;
```

Comment:
We don't use bare `if` checks anywhere else in the codebase. I'm good with it but it should be at least indented down.

```C#
if (DebugLogger.CheckLogLevel(LogLevel.Debug) && this._requestUri != value)
    requestUriIsModified = true;
```